sympy/physics/units/tests/test_quantities.py

```diff
@@ -206,6 +206,11 @@ def test_units():
     t = Symbol('t')
     assert integrate(t*m/s, (t, 1*s, 5*s)) == 12*m*s
     assert (t * m/s).integrate((t, 1*s, 5*s)) == 12*m*s
+    
+def test_eval_simplify():
+    assert ((cm/mm).simplify()) == 10
+    assert ((km/m).simplify()) == 1000
+    assert ((km/cm).simplify()) == 100000
```

Comment:
you should add some tests which causes an exception to be raised as well (and make sure the exception is of the right kind using SymPy's contextmangager `raises`).

Reply:
@bjodah I think the exceptions will be raised if any one of arguments of self will have `scale_factor` value as none or 0

Reply:
@bjodah I gave the value of `scale_factor` as `None` and it raised an `AttributeError`, I think this will work